### ISSUE TITLE
HCPC and ICD10 codes without mapping in current vocabulary

### ISSUE THREAD

**Issue by deppensa** [OHDSI/Vocabulary-v5.0#75]
Attached is a spreadsheet with HCPC, CPT and ICD-10 codes that are missing in current vocabulary.  Some of these codes are no longer in use (HCPC, CPT) or need to be looked at due to no corresponding map, ICD10. List is top 1000 codes from VINCI OMOP project and contain no PHI
[OMOPv5 tracking counts.xlsx](https://github.com/OHDSI/Vocabulary-v5.0/files/218014/OMOPv5.tracking.counts.xlsx)

**Comment by cgreich** [comment]
@deppensa:

These are codes deprecated long ago. For example, G0001 died in 2004, D7110 in 2002. We are happy to add them, but we'd need you to find us the description for them (you can google them one by one) and the date of deprecation.

Best,
C

**Comment by ericaVoss** [comment]
Can we close this issue?

**Comment by cgreich** [comment]
@ericaVoss: No, we need to fix it. Let me send an email to @deppensa, he/she might not pay attention.

@dimshitc: This is the problem of old HCPCS and CPT4s, which are no longer in UMLS. You may want to send an email to Tammy (she also does UMLS). Why there are ICD10 (probably ICD10CM) I don't know. We should have them all. Can you look?

**Comment by deppensa** [comment]
The VA OMOP group has moved to installing OMOP updates (v5+) and cleaning missing LOINC and drug codes as higher priority.  I don't know when we will get back to working on the list of old activity code mappings.

**Comment by ericaVoss** [comment]
Any progress?  /*poke*/

**Comment by p-talapova** [comment]
This issue is quite old, but we should solve it. 

@cgreich , this is about ICD10PCS, not ICD10 or ICD10CM.

We have analyzed the table of "OMOPv5 tracking counts.xlsx" and can conclude that there are: 
- 287 Inactive HCPCS and CPT4 codes that cannot be taken from anywhere
- 705 already Standard concepts  (ICD10PCS and HCPCS)
- 7 concepts without mappings (4 HCPCS non-standard, but valid codes, 3 ICD10PCS Hierarchical deprecated concepts)
So, we can map the latest during future refreshes. 

Note, that some concepts have the inappropriate domain for procedure_occurence table. They marked by yellow.

[HCPCS, ICD10PCS and CPT4 unmapped concept issue.xlsx](https://github.com/OHDSI/Vocabulary-v5.0/files/3858285/HCPCS.ICD10PCS.and.CPT4.unmapped.concept.issue.xlsx)

Can we close the issue?

**Comment by cgreich** [comment]
@p-talapova :

> 287 Inactive
Meaning, you can't find them anywhere? 

> inappropriate domain for procedure_occurence table. They marked by yellow.
Understood. Folks don't know where to put them if they are not mapped or Standard Concepts. And then they put everything HCPCS into the PROCEDURE_OCCURRENCE table as the best guess, since the P in HCPCS stands for "Procedure".

> Can we close the issue?
You can, unless you want to keep it open for the 287. Or are you saying these cannot be found? Google knows them.

**Comment by p-talapova** [comment]
@cgreich,  do we have to add these Inactive concepts? In such a case we can do it by googling each concept separately because there are no full lists of the codes with names and effective time in one place. 
For HCPCS, there is another way to pick the concepts - manually from here: https://www.cgsmedicare.com/jc/pubs/pdf/appendix_a.pdf

I thought that they were terminated many years ago and are not used anymore.
For example,
http://www.ic.nc.gov/ncic/pages/item-c.htm (CPT4)
http://www.icd9data.com/HCPCS/2007/K/K0538.htm 
http://www.icd9data.com/HCPCS/2006/L/L5664.htm

**Comment by cgreich** [comment]
Yes. Find them, add them, and then declare victory. Of course, more of these corpses are going to be washed ashore over time. Say that as well and encourage the community to contribute.

Something like:

1) We did them all
2) If you find more let us know.

**Comment by p-talapova** [comment]
@cgreich, thanks for clarifying this!

**Comment by mik-ohdsi** [comment]
very brave, @m-khitrun , very brave indeed!

**Comment by TinyRickC137** [comment]
@m-khitrun could we declare a victory already? We can open new issues if needed.

**Comment by m-khitrun** [comment]
Hello!
Indeed, Google still remembers some of these codes. 
My search is in progress, but I have already found a bunch of CPT4 codes. You can have a look at them in the attached file: 
[CPT4_codes.xlsx](https://github.com/user-attachments/files/16657874/CPT4_codes.xlsx)
We plan to add them during the next release.

**Comment by m-khitrun** [comment]
Hi all! We've added a part of these historical codes, the others could not be found. Thank you!
